node/src/main/java/CordaCaplet.java

```
@@ -80,7 +78,7 @@ protected String getVarValue(String var) {
         if (ATTR_APP_CLASS_PATH == attr) {
             T cp = super.attribute(attr);
 
-            (new File(baseDir, "cordapps")).mkdir();
+            (new File(baseDir, "cordapps")).mkdir(); // TODO: shall we check if the folder has been created?
```

I think it'd be worth just adding it. Something that simply tries to create the directory and then if the directory doesn't exist throws an exception. It would indicate deeper problems if a node can't, say, write a directory, given how many Corda creates. Fail fast is a good thing to have.

At a bare minimum if you aren't willing to commit to that at least emit an error message if it doesn't exist after this mkdir.